File: ex34.py
from sys import exit 


def create_char():
  print("what is your name?")
  name = input("> ")
  print("What is your gender?")
  print("M or F?")
  gender = input("> ")
  print("and your age is?")
  age = input("> ")
  print(f"so your name is {name}, your're a {gender}, and your {age} yesrs old.")
  print("what style fighter are you"?)
  print("Mage, Knight, Monk, or Ranger")
  style = input("> ")
  char = [name, gender, age, style,]

def gold_room():
  print("This room is full of gold. How much do you take?")

  choice = int(input("> "))
  if choice == 1 or choice == 0:
    dead("Man, learn to type a number.")
  elif choice < 50:
    print(f"you wanted {choice} gold coins.")
    print("Nice, you're not greedy, you win!")
    exit(0)
  else: 
    print(f"you wanted {choice}...")
    dead("You greedy bastard!")


def bear_room():
  print("There is a bear in here.")
  print("The bear has a bunch of honey.")
  print("The fat bear is in front of another door.")
  print("How are you going to move the bear?")
  bear_moved = False

  while True:
    choice = input("> ")

    if choice == "take honey":
      dead("The bear looks at you, then slaps your face off")
    elif choice == "taunt bear" and not bear_moved:
      print("The bear has moved from the door.")
      print("You can go through it now.")
      bear_moved = True
    elif choice == "taunt bear" and bear_moved:
      print("The bear has moved from the door.")
      print("You can go through it now.")
      bear_moved = True
    elif choice == "taunt bear" and bear_moved:
      dead("The bear gets pissed off and chews your leg off")
    elif choice == "open door" and bear_moved:
      gold_room()
    else: 
      print("I got no idea what that means.")

def cthulu_room():
  print("Here you see the great evil Cthulu.")
  print("He, it, whatever stares at you and you go insane.")
  print("Do you flee for your life or eat your head?")

  choice = input("> ")

  if 'flee' in choice:
    game()
  elif 'head' in choice:
    dead("well that was tasty")
  else: 
    cthulu_room()

def dead(why):
  print(why, "good job!")
  exit(0)

def game():
  print("You are in a dark room.")
  print("There is a door to your right and left.")
  print("Which do you take?")
  choice = input("> ")

  if choice == "left":
    bear_room()
  elif choice == "right":
    cthulu_room()
  else:
    dead("You stuble around the room until you starve")

def intro():
  print("welcome to my game!")
  print("1. to play")
  print('2. to exit')
  choice = int(input("> "))
  if choice == 1:
    game()
  elif choice == 2:
    exit(0)


intro()

File: ex34.rb
def dead
  puts "You died. Good Job"
  exit(0) 
end

def gold_room 
  puts "this is a room full of gold. How much do you take?"
  choice = $stdin.gets.strip
  if choice > 50
    dead
  else
    puts "you wanted #{choice} gold coins."
    puts "Nice, you're not greedy, you win!"
    exit(0)
  end 
end

def bear_room 
  puts "There is a bear in here."
  puts "The bear has a bunch of honey."
  puts "The fat bear is in front of another door."
  puts "How are you going to move the bear?"
  bear_moved = false 

  choice = $stdin.gets.strip 
  if choice === 'Take honey'
     dead
  elsif choice === 'taunt bear' && bear_moved === false 
    puts "The bear has moved from the door."
    puts "You can go through it now."
    bear_moved = true
  elsif choice === 'taunt bear' &&  bear_moved === true 
    dead
  elsif choice === "open door" && bear_moved === true 
    gold_room
  elsif choice === "open door" && bear_moved === false
    dead
  else 
    puts "I don't understand"
    bear_room
  end
end

def cthulu_room 
  puts "Here you see the great evil Cthulu.\nHe, it, whatever stares at you and you go insane.\nDo you flee for your life or eat your head?"
  puts "flee?"
  puts "eat head"
  choice = $stdin.gets.strip 
  if choice === 'flee'
    game
  elsif choice === 'eat head'
    dead
  else 
    cthulu_room
  end 
end

def game
  puts "You are in a dark room."
  puts "There is a door to your right and left."
  puts "Which do you take?"
  choice = $stdin.gets.strip
  if choice === 'left'
    bear_room
  elsif choice === 'right'
    cthulu_room
  else
    puts 'you stumble around in the dark and fall on a knife'
  end
end

game
